src/class/impl_.rs

```diff
@@ -795,3 +800,34 @@ pub(crate) unsafe extern "C" fn fallback_new(
 pub(crate) unsafe extern "C" fn tp_dealloc<T: PyClass>(obj: *mut ffi::PyObject) {
     crate::callback_body!(py, T::Layout::tp_dealloc(obj, py))
 }
+
+pub(crate) unsafe extern "C" fn sq_item_from_mapping(
```

Comment:
I know this is similar name to the original, but maybe we should use a more clear naming ? `seq` would make more sense than `sq`. overall I'd go for `sequence` but it might be long for some peeps. same for below